test/e2e/framework/cluster.go

```diff
@@ -0,0 +1,330 @@
+// Copyright (C) 2024 ScyllaDB
+
+package framework
+
+import (
+	"context"
+	"crypto/sha512"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	scyllaclientset "github.com/scylladb/scylla-operator/pkg/client/scylla/clientset/versioned"
+	"github.com/scylladb/scylla-operator/pkg/controllerhelpers"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/client-go/discovery"
+	cacheddiscovery "k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+const (
+	defaultUserAgent = "scylla-operator-e2e"
```

Comment:
`default` kind of implies this can be changed but it's hardcoded at some places

Reply:
I think this should actually be coming from the propagated rest config. Looks like atm we propagate them but then override with the default values anyway, so the flags don't actually change anything?

Reply:
/shrug

Reply:
removed the defaults, let me know if this should be a part of a separate PR